include/utils.h

```diff
@@ -90,4 +96,19 @@ std::string simplify_task_name(const std::string& demangled_type_name);
 // escapes "<", ">", and "&" with their corresponding HTML escape sequences
 std::string escape_for_dot_label(std::string str);
 
+template <typename... FmtParams>
+[[noreturn]] void throw_error(FmtParams&&... fmt_args) {
+	throw std::runtime_error(fmt::format(std::forward<FmtParams>(fmt_args)...));
+}
+
+template <typename... FmtParams>
+void report_error(const error_policy policy, FmtParams&&... fmt_args) {
+	switch(policy) {
+	case error_policy::ignore: break;
+	case error_policy::log_warning: CELERITY_WARN(std::forward<FmtParams>(fmt_args)...); break;
+	case error_policy::log_error: CELERITY_ERROR(std::forward<FmtParams>(fmt_args)...); break;
+	case error_policy::throw_exception: throw_error(std::forward<FmtParams>(fmt_args)...); break;
+	}
+}
```

Comment:
We talked about this in person, but
 - should we have both `fatal` and `throw_exception`, or should this be one call which throws in tests and aborts otherwise
 - if we make it one method, how would we detect that we are inside a test?
 - if there are two, should the default be `fatal` or still `throw_exception` in the policy structs of `task_manager` / `distributed_graph_generator`?